lib/auth/sessions_test.go

```diff
@@ -83,3 +84,35 @@ func TestServer_CreateWebSessionFromReq_deviceWebToken(t *testing.T) {
 		}
 	})
 }
+
+func Test_guessAWSRoleSessionName(t *testing.T) {
+	localClusterName := "teleport.abigcompany.com"
+
+	for _, tt := range []struct {
+		request  NewAppSessionRequest
+		expected string
+	}{
+		{
+			request: NewAppSessionRequest{
+				NewWebSessionRequest: NewWebSessionRequest{
+					User: "raimundo.oliveira@abigcompany.com",
+				},
+				ClusterName: "teleport.abigcompany.com",
+			},
+			expected: "raimundo.oliveira@abigcompany.com",
```

Comment:
Should we test with a local username that gets hashed too, in addition to this one?